solutions/account-infrastructure-base/variables.tf

```diff
@@ -102,3 +102,284 @@ variable "existing_cos_resource_group_name" {
+  type        = string
+  description = "The name to give the newly provisioned COS bucket which will be used for Activity Tracker logs, required if 'var.provision_atracker_cos' is true."
+  default     = null
+}
+
+variable "cos_bucket_access_tags" {
+  type        = list(string)
+  description = "A list of Access Tags applied to the created bucket."
+  default     = []
+}
+
+variable "cos_bucket_expire_enabled" {
+  type        = bool
+  description = "A flag to control expiry rule on the bucket."
+  default     = false
+}
+
+variable "cos_bucket_expire_days" {
+  type        = number
+  description = "Number of days before expiry."
+  default     = 365
+}
+
+variable "cos_bucket_object_versioning_enabled" {
+  type        = bool
+  description = "A flag to control object versioning on the bucket."
+  default     = false
+}
+
+variable "cos_bucket_storage_class" {
+  type        = string
+  description = "COS Bucket storage class type"
+  default     = null
+}
+
+variable "cos_bucket_archive_enabled" {
+  type        = bool
+  description = "Set as true to enable archiving on the COS bucket."
+  default     = false
+}
+
+variable "cos_bucket_archive_days" {
+  type        = number
+  description = "Number of days to archive objects in the bucket."
+  default     = 20
+}
+
+variable "cos_bucket_archive_type" {
+  type        = string
+  description = "Type of archiving to use on bucket."
+  default     = "Glacier"
+}
+
+variable "cos_bucket_retention_enabled" {
+  type        = bool
+  description = "Retention enabled for COS bucket."
+  default     = false
+}
+
+variable "cos_bucket_retention_default" {
+  description = "Specifies default duration of time an object that can be kept unmodified for COS bucket."
+  type        = number
+  default     = 90
+}
+
+variable "cos_bucket_retention_maximum" {
+  description = "Specifies maximum duration of time an object that can be kept unmodified for COS bucket."
+  type        = number
+  default     = 350
+}
+
+variable "cos_bucket_retention_minimum" {
+  description = "Specifies minimum duration of time an object must be kept unmodified for COS bucket."
+  type        = number
+  default     = 90
+}
+
+variable "cos_bucket_retention_permanent" {
+  description = "Specifies a permanent retention status either enable or disable for COS bucket."
+  type        = bool
+  default     = false
+}
+
+variable "cos_bucket_cbr_rules" {
+  type = list(object({
+    description = string
+    account_id  = string
+    rule_contexts = list(object({
+      attributes = optional(list(object({
+        name  = string
+        value = string
+      })))
+    }))
+    enforcement_mode = string
+    tags = optional(list(object({
+      name  = string
+      value = string
+    })), [])
+    operations = optional(list(object({
+      api_types = list(object({
+        api_type_id = string
+      }))
+    })))
+  }))
+  description = "COS Bucket CBR Rules"
+  default     = []
+}
+
+variable "cos_instance_cbr_rules" {
```

Comment:
don't expose this, support is coming in https://github.com/terraform-ibm-modules/terraform-ibm-account-infrastructure-base/pull/86